backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs

```diff
@@ -104,9 +107,16 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
             }
 
             string linkJson = await req.ReadAsStringAsync();
-            AssignmentLinkDto linkDto = JsonConvert.DeserializeObject<AssignmentLinkDto>(linkJson);
+            AssignmentLinkDto linkDto;
+            try
+            {
+                linkDto = JsonConvert.DeserializeObject<AssignmentLinkDto>(linkJson);
+            } catch (Exception ex)
+            {
+                return new BadRequestErrorMessageResult("Could not create a valid link from the request. " + ex.Message);
+            }
 
-            if (linkId != linkDto.Id)
+            if (linkId != linkDto.Id.ToString())
```

Comment:
What would it take to start accepting `linkId` as **Guid** also?

Reply:
The row key in an azure table can only be a string; we cannot override this explicitly, which is why we are having to put this validation in the dto.